0015-estimator-interface.md

```diff
@@ -0,0 +1,457 @@
+We propose that the internal data model of this object have an element format
+
+```python
+List[Tuple[str, List[int], float]]
+```
+
+where, for example, the observable `0.5 * IIZY + 0.2 * XIII` would be stored as
+
+```python
+[("ZY", [2, 3], 0.5), ("X", [0], 0.2)]
+```
+
+This is proposed instead of `{"IIZY": 0.5, "XIII": 0.2}` anticipating the overhead of storing and working with so many `"I"`'s for large devices.
+
+Note that the above is simply the data model; the `ObservablesArray` object, either in the constructor or special static coercion method, will accept the `quantum_info` objects such as `Pauli` and `SparsePauliOp` that they are used to.
+Further, through [type coersion](#type-coercion-strategy) in `Estimator.run()` itself, a user won't need to manually construct an `ObservablesArray`, as shown in the example in the [Design Proposal](#design-proposal) section.
+
+It is up to each `Estimator` implementation to, if it sees fit, to solve a graph coloring problem that decides how to produce a sufficient set of physical circuits that are capable of producing data to make each of these estimates.
+
+### Arrays and Broadcasting <a name="arrays-and-broadcasting"></a>
+
+An `nd-array` is an object whose elements are indexed by a tuple of integers, where each integer must be bounded by the dimension of that axis. 
+The tuple of dimensions, one for each axis, is called the shape of the `nd-array`. 
+For example, 1D list of 10 objects (typically numbers) is a vector and has shape `(10,)`; a matrix with 5 rows and 2 columns is 2D and has shape `(5, 2)`; a single number is 0D and has an empty shape tuple `()`; an `nd-array` with shape (20, 10, 5) can be interpreted as a length-20 list of 10×5 matrices.
+
+We propose the constraint that a `BindingsArray` instance and an `ObservablesArray` instance that live in the same task must have shapes that are broadcastable, in which case, the `Estimator` promises to return one expectation value estimate for each element of the broadcasted shape.
+
+Here are some examples of common patterns expressed in terms of array broadcasting, and their accompanying visual representation in the figure that follows:
+
+```python
+# Broadcast single observable
+parameter_values = np.array(1.0)
+parameter_values.shape == ()
+observables = ObservablesArray(["III", "XXX", Pauli("YYY"), "ZZZ", Pauli("XYZ")])
+observables.shape == (5,)
+>> result_bundle.shape == (5,)
+
+# Inner/Zip
+parameter_values = BindingsArray(np.random.uniform(size=(5,)))
+parameter_values.shape == (5,)
+observables = ObservablesArray([[Pauli("III")], [Pauli("XXX")], [Pauli("YYY")], [Pauli("ZZZ")], [Pauli("XYZ")]])
+observables.shape == (5,1)
+>> result_bundle.shape == (5,)
+
+# Outer/Product
+parameter_values = BindingsArray(np.random.uniform(size=(1,6)))
+parameter_values.shape == (1,6)
+observables = ObservablesArray([[Pauli("III"), Pauli("XXX"), Pauli("YYY"), Pauli("ZZZ"), Pauli("XYZ"), Pauli("IIZ")]])
```

Comment:
Shouldn't they be 4 observables? 
```suggestion
observables = ObservablesArray([[Pauli("III"), Pauli("XXX"), Pauli("YYY"), Pauli("ZZZ")]])
```

Reply:
the shape (4,1) shouldn't be link this?
```suggestion
observables = ObservablesArray([[Pauli("III")], [Pauli("XXX")], [Pauli("YYY")], [Pauli("ZZZ")]])
```

Reply:
I agree with @1ucian0

Reply:
Thanks, hopefully all of these typos have been fixed in 20859ef